dds/DCPS/PrinterValueWriter.h

```diff
@@ -0,0 +1,290 @@
+  at_newline_ = false;
+}
+
+void PrinterValueWriter::end_discriminator()
+{}
+
+void PrinterValueWriter::begin_union_member(const char* name)
+{
+  stream_ << newline() << std::string(current_indent_, ' ') << name << ": ";
+  at_newline_ = false;
+}
+
+void PrinterValueWriter::end_union_member()
+{}
+
+void PrinterValueWriter::begin_array()
+{
+  current_indent_ += indent_;
+}
+
+void PrinterValueWriter::end_array()
+{
+  current_indent_ -= indent_;
+}
+
+void PrinterValueWriter::begin_sequence()
+{
+  current_indent_ += indent_;
+}
+
+void PrinterValueWriter::end_sequence()
+{
+  current_indent_ -= indent_;
+}
+
+void PrinterValueWriter::begin_element(size_t idx)
+{
+  stream_ << newline() << std::string(current_indent_, ' ') << '[' << idx << "]: ";
+  at_newline_ = false;
+}
+
+void PrinterValueWriter::end_element()
+{}
+
+void PrinterValueWriter::write_boolean(ACE_CDR::Boolean value)
+{
+  stream_ << (value ? "true" : "false");
+}
+
+void PrinterValueWriter::write_byte(ACE_CDR::Octet value)
+{
+  stream_ << std::hex << "0x" << static_cast<unsigned int>(value) << std::dec;
+}
+
+#if OPENDDS_HAS_EXPLICIT_INTS
+
+void PrinterValueWriter::write_int8(ACE_CDR::Int8 value)
+{
+  stream_ << static_cast<int>(value);
+}
+
+void PrinterValueWriter::write_uint8(ACE_CDR::UInt8 value)
+{
+  stream_ << static_cast<unsigned int>(value);
+}
+#endif
+
+void PrinterValueWriter::write_int16(ACE_CDR::Short value)
+{
+  stream_ << value;
+}
+
+void PrinterValueWriter::write_uint16(ACE_CDR::UShort value)
+{
+  stream_ << value;
+}
+
+void PrinterValueWriter::write_int32(ACE_CDR::Long value)
+{
+  stream_ << value;
+}
+
+void PrinterValueWriter::write_uint32(ACE_CDR::ULong value)
+{
+  stream_ << value;
+}
+
+void PrinterValueWriter::write_int64(ACE_CDR::LongLong value)
+{
+  stream_ << value;
+}
+
+void PrinterValueWriter::write_uint64(ACE_CDR::ULongLong value)
+{
+  stream_ << value;
+}
+
+void PrinterValueWriter::write_float32(ACE_CDR::Float value)
+{
+  stream_ << value;
+}
+
+void PrinterValueWriter::write_float64(ACE_CDR::Double value)
+{
+  stream_ << value;
+}
+
+void PrinterValueWriter::write_float128(ACE_CDR::LongDouble value)
+{
+  stream_ << value;
+}
+
+void PrinterValueWriter::write_fixed(const OpenDDS::FaceTypes::Fixed& /*value*/)
+{
+  // FUTURE
+  stream_ << "fixed";
+}
+
+void PrinterValueWriter::write_char8(ACE_CDR::Char value)
+{
+  if (std::isprint(static_cast<unsigned char>(value))) {
+    stream_ << "'" << value << "'";
+  } else {
+    stream_ << "'\\" << std::oct << static_cast<unsigned int>(value) << std::dec << "'";
+  }
+}
+
+void PrinterValueWriter::write_char16(ACE_CDR::WChar value)
+{
+  // FUTURE
+  stream_ << "wide character";
+}
+
+void PrinterValueWriter::write_string(const ACE_CDR::Char* value, size_t length)
+{
+  stream_ << '"' << std::string(value, length) << '"';
+}
+
+void PrinterValueWriter::write_wstring(const ACE_CDR::WChar* /*value*/, size_t /*length*/)
+{
+  // FUTURE
+  stream_ << "wide string";
+}
```

Comment:
There are existing functions in [`ValueHelper.h`](https://github.com/objectcomputing/OpenDDS/blob/master/dds/DCPS/ValueHelper.h) for printing a representation all the character based types. These are the same functions `opendds_idl` and `print_dynamic_data`/`inspect` are using.